wideq/dryer.py

```diff
@@ -0,0 +1,205 @@
+    ERROR_TE1 = '@WM_US_DRYER_ERROR_TE1_W'
+    ERROR_TE2 = '@WM_US_DRYER_ERROR_TE2_W'
+    ERROR_TE5 = '@WM_US_DRYER_ERROR_TE5_W'
+    ERROR_TE6 = '@WM_US_DRYER_ERROR_TE6_W'
+    UNKNOWN = _UNKNOWN
+
+
+class TempControl(enum.Enum):
+    """Represents temperature control setting."""
+
+    OFF = '-'
+    ULTRA_LOW = '@WM_DRY27_TEMP_ULTRA_LOW_W'
+    LOW = '@WM_DRY27_TEMP_LOW_W'
+    MEDIUM = '@WM_DRY27_TEMP_MEDIUM_W'
+    MID_HIGH = '@WM_DRY27_TEMP_MID_HIGH_W'
+    HIGH = '@WM_DRY27_TEMP_HIGH_W'
+    UNKNOWN = _UNKNOWN
+
+
+class TimeDry(enum.Enum):
+    """Represents a timed dry setting."""
+
+    OFF = '-'
+    TWENTY = '20'
+    THIRTY = '30'
+    FOURTY = '40'
+    FIFTY = '50'
+    SIXTY = '60'
+    UNKNOWN = _UNKNOWN
+
+
+class DryerDevice(Device):
+    """A higher-level interface for a dryer."""
+
+    def poll(self) -> Optional['DryerDevice']:
+        """Poll the device's current state.
+
+        Monitoring must be started first with `monitor_start`.
+
+        :returns: Either a `DryerStatus` instance or `None` if the status is
+            not yet available.
+        """
+        # Abort if monitoring has not started yet.
+        if not hasattr(self, 'mon'):
+            return None
+
+        res = self.mon.poll_json()
+        if res:
+            return DryerStatus(self, res)
+        else:
+            return None
+
+
+class DryerStatus(object):
+    """Higher-level information about a dryer's current status.
+
+    :param dryer: The DryerDevice instance.
+    :param data: JSON data from the API.
+    """
+
+    def __init__(self, dryer: DryerDevice, data: dict):
+        self.dryer = dryer
+        self.data = data
+
+    def get_bit(self, key: str, index: int) -> str:
+        bit_value = int(self.data[key])
+        bit_index = 2 ** index
+        mode = bin(bit_value & bit_index)
+        if mode == bin(0):
+            return 'OFF'
+        else:
+            return 'ON'
+
+    @property
+    def state(self) -> DryerState:
+        """Get the state of the dryer."""
+        attr = 'State'
+        return DryerState(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def previous_state(self) -> DryerState:
+        """Get the previous state of the dryer."""
+        attr = 'PreState'
+        return DryerState(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def dry_level(self) -> DryLevel:
+        """Get the dry level."""
+        attr = 'DryLevel'
+        return DryLevel(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def temperature_control(self) -> TempControl:
+        """Get the temperature control setting."""
+        attr = 'TempControl'
+        return TempControl(self.dryer.model.enum_name(attr, self.data[attr]))
+
+    @property
+    def time_dry(self) -> TimeDry:
+        """Get the time dry setting."""
+        attr = 'TimeDry'
+        return TimeDry(self.dryer.model.enum_name(attr, self.data[attr]))
```

Comment:
Maybe a helper method like this would be useful?

```py
def _lookup_enum(self, attr):
    return self.dryer.model.enum_name(attr, self.data[attr])
```

Reply:
That makes sense.